Tribes/src/app/_components/building-type/building-type.component.html

```diff
@@ -1,87 +1,87 @@
-        </div>
-        <div *ngIf="towhalllevel !== 3 || goldAmount > getUpgradingCost(towhalllevel)" class="text-center mt-3">
-          <button class="btn btn-warning">
-            <p class="m-0">Upgrade</p>
-            <small>for {{getUpgradingCost(townhallLevel)}} gold</small>
-          </button>
-        </div>
-      </div>
-      <div *ngIf="type !== 'townhall'">
-        <div class="row align-items-center justify-content-center mb-5">
-          <img src="assets/images/{{type}}.png" class="main-image" />
-          <div class="h2"> Details of your {{type}}s</div>
-        </div>
-        <div *ngFor="let level of levels" class="row mt-5 align-items-center">
-          <div class="col-2 level-icons">
-            <div *ngFor="let i of [].constructor(level)" class="mx-auto text-center image-position">
-              <img src="assets/images/{{type}}.png" class="level-image" />
-            </div>
+    <div>
+      <app-header></app-header>
+    </div>
+    <div class="row">
+      <div class="col-md-8">
+        <button class="btn btn-danger back" onclick="window.location.href ='kingdom'">Back</button>
+        <div *ngIf="type === 'townhall'">
+          <div class="row align-items-center justify-content-center mb-5">
+            <img src="assets/images/{{type}}.png" class="main-image" />
+            <div class="h2">Details of your {{type}}</div>
           </div>
-          <div class="col-md-4">
-            <span *ngIf="getNumberOfBuildingsByLevel(level) > 0">{{getNumberOfBuildingsByLevel(level)}} {{type}}(s) of
-              level {{level}}</span>
-            <span *ngIf="getNumberOfBuildingsByLevel(level) == 0">You don't have any {{type}} of level {{level}}</span>
+          <div class="row mt-5 justify-content-center align-items-center">
+            <p class="h3">Your townhall is at level {{townhallLevel}}.</p>
           </div>
-          <div class="col-md-3">
-            <span *ngIf="type !== 'barrack'">Production: {{getProductionRate(level)}}/min.</span>
-            <span *ngIf="type === 'barrack'">Training: {{level}} soldier(s) at time</span>
+          <div class="row mt-3 justify-content-center align-items-center">
+            <p>Capacity: {{getTownhallFoodCapacity(townhallLevel)}} food and {{getTownhallGoldCapacity(townhallLevel)}}
+              gold</p>
           </div>
-          <div class="col-md-3"
-            *ngIf="level !== 3 && !checkIfBuildingIsUpgradeable(level)">
+          <div *ngIf="towhalllevel !== 3 && getUpgradingCost(townhallLevel) > goldAmount" class="text-center mt-3">
             <button class="btn btn-warning inactive">
               <p class="m-0">Upgrade</p>
               <small>for {{getUpgradingCost(level)}} gold</small>
             </button>
           </div>
-          <div class="col-md-3"
-            *ngIf="level !== 3 && checkIfBuildingIsUpgradeable(level)">
+          <div *ngIf="towhalllevel !== 3 || goldAmount > getUpgradingCost(towhalllevel)" class="text-center mt-3">
```

Comment:
When we have these more complex ifs around, we rather put them in the code and name them instead of hardcoding it into the template.